### ISSUE TITLE
Opening python file returns Error detected while processing FileType Autocommands for "*": E5108: Error executing Lua

### ISSUE THREAD

**Issue by sjeon7** [nvim-treesitter/nvim-treesitter#526]
**Describe the bug**
Opening python file returns the error message below.

```
Error detected while processing FileType Autocommands for "*":
E5108: Error executing lua ...sr/share/nvim/runtime/lua/vim/treesitter/highlighter.lua:76: attempt to index a nil value
```
Opening python files with this error has no highlights.

Moving the cursor when opening python file also creates the error message below.
```
Error detected while processing CursorHold Autocommands for "<buffer=1>":
E5108: Error executing lua ...plugged/nvim-treesitter/lua/nvim-treesitter/ts_utils.lua:119: attempt to index a nil value
```

**To Reproduce**
1. With Neovim version 0.5.0-709-g4f8d98e58 (latest nightly)
2. With clean init.vim
```
call plug#begin('~/.config/nvim/plugged')
        Plug 'nvim-treesitter/nvim-treesitter'
call plug#end()

lua << EOF
require'nvim-treesitter.configs'.setup {
        highlight = {
                enable = true,
        },
        refactor = {
                highlight_definitions = {
                        enable = true
                },
        },
        ensure_installed = 'python'
}
EOF
```
3. Plug update for newest nvim-treesitter and TSInstall python. 
4. TSUpdate python
5. Open any python file

**Expected behavior**
Python file opens without error messages. Navigating the python code also should not create error messages.

**Output of `:checkhealth nvim_treesitter`**

<details>
<code>

```
## python parser healthcheck
  - OK: python parser found.
  - OK: `highlights.scm` found.
  - ERROR: Error in `locals.scm` query found for python
    ...LpHx/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:20: query: invalid node type at position 1421
    - ADVICE:
      - Try `:TSUpdate python` (queries might have been adapted for a upstream parser change)
      - Try to find the syntax error/invalid node type in above file.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.
```
</code>
</details>

**Output of `nvim --version`**
```
NVIM v0.5.0-709-g4f8d98e58
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENER
ATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -
I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-6c0c5b42-8f4c-41ab-9dd7-cb182e87f565

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/travis/build/neovim/bot-ci/build/neovim/build/nvim.AppDir/usr/share/nvim
"
```
**Additional context**
This happens after updating nvim-treesitter using vim-plug and updating python query using TSUpdate python.
Before updates, nvim-treesitter has no conflict with python files.
Might be related to #254 and other similar issues, but updating/reinstalling python parser(using TSInstall/TSUpdate) or nvim-treesitter itself does not resolve this issue.

**Comment by theHamsta** [comment]
`:TSUpdate python` since #514 introduced `left_hand_side` node type. It was merged a bit early.

**Comment by kosayoda** [comment]
From the original issue, and from what I am also experiencing, updating does not fix the issue.

I'm on the latest neovim nightly, treesitter and python parser.

**Comment by theHamsta** [comment]
The problems in locals.scm can be solved when out-commenting:
```lisp
; not a scope!
;(for_statement
  ;left: (variables
          ;(identifier) @definition.var))

; not a scope!
;(while_statement) @scope

; for in list comprehension
;(for_in_clause
  ;left: (variables
          ;(identifier) @definition.var))

```
apparently `left_hand_side` is used there now.

**Comment by sjeon7** [comment]
Reverting back to tree-sitter-python parser to https://github.com/tree-sitter/tree-sitter-python/tree/30f538aa1f932a2864c102694761d3127a61dc68 might solve this error temporarily.

This cannot be the solution but it can be used without error before the update for this bug fixed.

 `git clone https://github.com/tree-sitter/tree-sitter-python.git `
`cd tree-sitter-python`
`git reset --hard 30f538aa1f932a2864c102694761d3127a61dc68`
`gcc -o parser.so -I./src src/parser.c src/scanner.cc -shared -Os -lstdc++ -fPIC`
`mv parser.so ~/.config/nvim/plugged/nvim-treesitter/parser/python.so`

Again, this is only temporary solution and hope that the new update come out soon.
Thank you for the great plugin by the way!

**Comment by sanga** [comment]
Thank you for figuring that out @sjeon7 ! Small note that I guess the final path should actually be: `mv parser.so ~/.local/share/nvim/plugged/nvim-treesitter/parser/python.so` (it is for me at least and I think that is the correct default path)

**Comment by sjeon7** [comment]
@sanga Thank you for the note. I edited the comment :)

**Comment by ray-x** [comment]
Might be irrelevant, but treesitter for `golang` also failed to load with E5108 error with latest version.

**Comment by theHamsta** [comment]
I assume the parser had a runtime update and require a newer tree-sitter runtime. I'm on mobile so I haven't checked.

**Comment by alexaandru** [comment]
Came here to report the same error, but for Go.

I used a tiny `test.vim` config:

```
packadd nvim-lspconfig
lua require'nvim_lsp'.gopls.setup{}

packadd nvim-treesitter
lua require'nvim-treesitter.configs'.setup{highlight = {enable = true}, ensure_installed = {'go'}}
```

then `nvim -u test.vim main.go` to test, where `main.go` is:

```Go
package main

func main() {
	println("Hello world")
}
```

My nvim is the very latest, v0.5.0-712 and the plugin is the latest master (I `git pull` it, don't use any plugins management systems, other than the builtin, `packadd`).

**Comment by theHamsta** [comment]
Could you test `:TSUpdate` with https://github.com/nvim-treesitter/nvim-treesitter/pull/351 (only after a backup of your compiled parsers)?

It introduces a lockfile that has newest revisions for all parsers except for python/go that did a runtime update.

**Comment by sjeon7** [comment]
@theHamsta Tested with https://github.com/nvim-treesitter/nvim-treesitter/pull/351, this resolves the issue for python!

For golang, I tested with simple golang code below and it seems to have no conflicts.
```
package main

import "fmt"

func main() {
    fmt.Println("Hello, World.")
}
```
If anyone can test golang, it's appreciated.

I guess this issue can be closed after new pr https://github.com/nvim-treesitter/nvim-treesitter/pull/351 is merged.

Thank you for your help!

**Comment by ray-x** [comment]
Verified golang with pr #351 . Works fine and does not see E5108.

**Comment by theHamsta** [comment]
Closed via #351

**Comment by alexaandru** [comment]
I see it's merged into master, so gonna test with latest master... and yes, it works for me too, for Go: no error and I see the nice syntax colors that only tree sitter can give :-) Thank you!! Cheers!
